### PR TITLE
Misc fixes.

### PR DESCRIPTION
**Pull request by fisx**

### CODE REVIEW

File: src/Frontend/Page/Idea.hs [hunk 1]
```diff
@@ -270,13 +270,12 @@ instance FormPage EditIdea where
...
-    -- FIXME: category choice should look like in CreateIdea.
-    formPage = createOrEditPage True
+    formPage v form p@(EditIdea idea) = createOrEditPage True (idea ^. ideaLocation) v form p
 
 createOrEditPage :: (Monad m, Typeable page, Page page) =>
-    Bool -> View (HtmlT m ()) -> (HtmlT m () -> HtmlT m ()) -> page -> HtmlT m ()
-createOrEditPage showDeleteButton v form p = semanticDiv p $ do
+    Bool -> IdeaLocation ->
+    View (HtmlT m ()) -> (HtmlT m () -> HtmlT m ()) -> page -> HtmlT m ()
+createOrEditPage showDeleteButton cancelUrl v form p = semanticDiv p $ do
```
Comment (andorp) [inline src/Frontend/Page/Idea.hs]: Is the `showDeleteButton` could be computed from `renderContext` via extending the `ideaCapabilities` description?

Comment (fisx) [inline src/Frontend/Page/Idea.hs]: yes, good idea.  will do!

Comment (fisx) [inline src/Frontend/Page/Idea.hs]: Actually, i changed my mind.  `showDeleteButton` is set by one of the two callers of this, which are the two `ToHtml` instances of `EditIdea`, `CreateIdea` resp.

So this is not really about the session context, but about which of two similar forms we happen to render.

Does that make sense?

Comment (andorp) [inline src/Frontend/Page/Idea.hs]: Yes. On the other hand I am agree with Connor McBride, we should introduce different types for these parameters, even if at the beginning it is isomorphic to `Bool`. But I can live with the current solution.

File: src/Frontend/Page/Topic.hs [hunk 2]
```diff
@@ -23,11 +23,12 @@ where
 import Action (ActionM, ActionPersist(..), ActionUserHandler, ActionExcept, renderContext)
 import Control.Exception (assert)
 import Frontend.Page.Category
+import Frontend.Page.Overview
```
Comment (andorp) [inline src/Frontend/Page/Topic.hs]: One page module should not include other one.We have an issue #346 connected to this change. But for the time being it is ok.

**Comment by andorp** [comment]
Two minor comments, otherwise LGTM

**Pull request merged**